Nomad: memory oversubscription supports reserve-only requests (#1629)
hashicorp/web-unified-docs: content/nomad/v1.11.x/content/docs/job-specification/resources.mdx

limits that correspond to the hard and soft limits described previously.

- `memory`: The hard limit unless `memory_max` is set. If `memory_max` is set,
  then `memory` is the reservation.
- `memory_max`: The hard limit if set. If `memory_max` is unset, then `memory`
  is the hard limit. If `memory_max` is set to `-1` and it is supported in the
  driver, the task has no hard limit.

The `memory_max` limit attribute is currently supported by the official
`raw_exec`, `exec2`, `exec`, `docker`, `podman`, and `java` task
drivers. Consult the documentation of community-supported task drivers for their
memory oversubscription support.

Memory oversubscription by setting `memory_max` is opt-in. Nomad operators can
enable [Memory Oversubscription in the scheduler
configuration][api_sched_config]. Enterprise customers can use [Resource
Quotas][quota_spec] to limit the memory oversubscription and enable or disable
Memory oversubscription by setting `memory_max` is opt-in. Nomad operators can
enable memory oversubscription in the scheduler configuration. Refer to the
[`MemoryOversubscritionEnabled` parameter][api_sched_config] for details.

> Enterprise customers can use [resource quotas][quota_spec] to limit the memory

Reviewer: ```suggestion
Memory oversubscription by setting `memory_max` is opt-in. Nomad operators can
enable memory oversubscription in the scheduler configuration. Refer to the
[`MemoryOversubscritionEnabled` parameter][api_sched_config] for details.
Enterprise customers can use [resource quotas][quota_spec] to limit the memory
```
same comment as in the 1.10 version

Setting task memory limits requires balancing the risk of interrupting tasks
against the risk of wasting resources. If a task memory limit is set too low,
the task may exceed the limit and be interrupted; if the task memory is too
high, the cluster is left underutilized. Nomad drivers configure two separate
memory limits:

* The _hard_ limit represents the maximum memory the task can use. The operating
  system typically terminates the task if the task exceeds this limit.
* The _soft_ limit represents a reserve to represent the task's typical memory
  usage. The behavior of the soft limit is operating system dependent. On Linux,

>   the operating system will allow memory usage above this limit but subject the

Reviewer: ```suggestion
  the operating system allows memory usage above this limit but subject the
```
present tense nit

  task to reclaim pressure.

If a client's memory becomes contended or low, the operating system will
pressure the running tasks to free up memory. If the contention persists, Nomad
may kill oversubscribed tasks and reschedule them to other clients. The exact
mechanism for memory pressure is specific to the task driver, operating system,
and application runtime.

To help maximize cluster memory utilization while allowing a safety margin for
unexpected load spikes, Nomad allows job authors to set two separate resource

> limits that correspond to the hard and soft limits described above.

Reviewer: ```suggestion
limits that correspond to the hard and soft limits described previously.
```
nit: "above" is positional so use "previously" in case screen reader

* `memory`: the hard limit unless `memory_max` is set. If `memory_max` is set,
  then `memory` is the soft limit.
* `memory_max`: the hard limit if set. If `memory_max` is unset, then `memory`
  is the hard limit. If `memory_max` is set to -1 and it is supported in the
  driver, the task has no hard limit.

The `memory_max` limit attribute is currently supported by the official
`raw_exec`, `exec2`, `exec`, `docker`, `podman`, and `java` task
drivers. Consult the documentation of community-supported task drivers for their
memory oversubscription support.

Memory oversubscription by setting `memory_max` is opt-in. Nomad operators can
enable [Memory Oversubscription in the scheduler
configuration][api_sched_config]. Enterprise customers can use [Resource
Quotas][quota_spec] to limit the memory oversubscription and enable or disable

> memory oversubscription per [node pool][np_sched_config].

Reviewer: ```suggestion
Memory oversubscription by setting `memory_max` is opt-in. Nomad operators can
enable memory oversubscription in the scheduler configuration. Refer to
the [`MemoryOversubscritionEnabled` parameter]][api_sched_config] for details.
Enterprise customers can use [resource quotas][quota_spec] to limit the memory
oversubscription and enable or disable memory oversubscription per [node
pool][np_sched_config].
```
add relevant parameter.
